packages/math-input/src/components/input/mathquill-helpers.ts

```diff
@@ -0,0 +1,271 @@
+import {
+    MathFieldActionType,
+    MathFieldCursor,
+    MathFieldInterface,
+} from "../../types";
+
+import {CursorContext} from "./cursor-contexts";
+import MQ from "./mathquill-instance";
+
+const Numerals = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9"];
+const GreekLetters = ["\\theta", "\\pi"];
+const Letters = [
+    "A",
+    "B",
+    "C",
+    "D",
+    "E",
+    "F",
+    "G",
+    "H",
+    "I",
+    "J",
+    "K",
+    "L",
+    "M",
+    "N",
+    "O",
+    "P",
+    "Q",
+    "R",
+    "S",
+    "T",
+    "U",
+    "V",
+    "W",
+    "X",
+    "Y",
+    "Z",
+];
+
+// We only consider numerals, variables, and Greek Letters to be proper
+// leaf nodes.
+const ValidLeaves = [
+    ...Numerals,
+    ...GreekLetters,
+    ...Letters.map((letter) => letter.toLowerCase()),
+    ...Letters.map((letter) => letter.toUpperCase()),
+];
+
+export function getCursor(mathField: MathFieldInterface): MathFieldCursor {
+    return mathField.__controller.cursor;
+}
+
+export function isFraction(node): boolean {
```

Comment:
These functions seemed like they could be useful on their own, without needing to be tied to a class. I also needed to split them out for the handlers.